k8s/client.go

```diff
@@ -53,7 +55,29 @@ type client struct {
 }
 
 // New creates a client for the kubernetes apiserver.
-func New(apiServerURL string, caCert []byte, token string) (Client, error) {
+func New(caCertFile string, tokenFile string, apiServer string) Client {
+	caCert := readFile(caCertFile)
+	token := string(readFile(tokenFile))
+
+	client, err := createK8sClient(apiServer, caCert, token)
+	if err != nil {
+		log.Errorf("Unable to create Kubernetes client: %v", err)
+		os.Exit(-1)
```

Comment:
Libraries should not be calling `os.Exit`. Only main.go should ever exit to the OS. Instead, return an error and let the main code handle it as it wants.